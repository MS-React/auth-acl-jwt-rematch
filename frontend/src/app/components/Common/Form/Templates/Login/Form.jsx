import React from 'react';
import PropTypes from 'prop-types';

import FormHOC from '../../FormHOC';
import FormInput from '../../Elements/Input';

const LoginForm = ({
  onSubmit, onInputChange, fields, errors,
}) => (
  <form id="login-form" onSubmit={onSubmit}>
    <FormInput
      inputId="name"
      label="Username or email address"
      onChange={onInputChange}
      type="text"
      name="name"
      placeholder="Username or email address"
      required={fields.name.required}
      value={fields.name.value}
      invalid={errors.name && true}
      feedback={errors.name}
    />
    <FormInput
      inputId="password"
      label="Password"
      onChange={onInputChange}
      type="password"
      name="password"
      placeholder="Password"
      value={fields.password.value}
      required={fields.password.required}
      invalid={errors.password && true}
      feedback={errors.password}
    />
    <span>
      Hint:
      <i>
        username/password
      </i>
    </span>
    <button
      className="btn btn-lg btn-primary btn-block"
      type="submit"
      // disabled={isSubmitDisabled ? 'disabled' : ''}
    >
      Sign in
    </button>
  </form>
);

LoginForm.propTypes = {
  fields: PropTypes.object.isRequired,
  errors: PropTypes.object,
  onSubmit: PropTypes.func.isRequired,
  onInputChange: PropTypes.func.isRequired,
};

LoginForm.defaultProps = {
  errors: {},
};

export default (FormHOC(LoginForm));
