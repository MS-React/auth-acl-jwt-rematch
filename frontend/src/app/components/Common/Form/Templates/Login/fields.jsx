export const fields = {
  name: {
    value: '',
    required: true,
    rules: [{ method: 'isEmpty' }],
  },
  password: {
    value: '',
    required: true,
    rules: [{ method: 'isEmpty' }],
  },
};
