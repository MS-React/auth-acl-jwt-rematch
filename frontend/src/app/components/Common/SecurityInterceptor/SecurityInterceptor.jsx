import React from 'react';
import PropTypes from 'prop-types';
import { Route } from 'react-router';

export default class SecurityInterceptor extends React.PureComponent {
  static propTypes = {
    children: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
    authenticate: PropTypes.func.isRequired,
    logged: PropTypes.bool,
    UnauthenticatedComponent: PropTypes.func,
    token: PropTypes.string,
  };

  static defaultProps = {
    logged: false,
    UnauthenticatedComponent: null,
    token: null,
  }

  componentWillMount() {
    this.props.authenticate(this.props.token);
  }

  render() {
    const {
      logged, UnauthenticatedComponent, children, ...rest
    } = this.props;

    if (!logged) {
      return <UnauthenticatedComponent key="unauthenticated-key" />;
    }

    return (
      <Route {...rest}>
        <React.Fragment>
          {children}
        </React.Fragment>
      </Route>
    );
  }
}
