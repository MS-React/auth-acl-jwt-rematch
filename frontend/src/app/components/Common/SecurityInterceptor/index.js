export { default } from './SecurityInterceptor.container';
