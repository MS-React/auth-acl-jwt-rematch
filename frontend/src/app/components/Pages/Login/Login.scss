.login-page {
  &--form {
    left: 50%;
    margin: auto;
    max-width: 330px;
    padding: 15px;
    position: absolute;
    text-align: center;
    top: 40%;
    transform: translate(-50%, -50%);
    width: 100%;

    .login-form {
      .form-control {
        box-sizing: border-box;
        font-size: 16px;
        height: auto;
        padding: 10px;
        position: relative;
      }
    
      .form-control:focus {
        z-index: 2;
      }
    
      input[type="email"] {
        border-bottom-left-radius: 0;
        border-bottom-right-radius: 0;
        margin-bottom: -1px;
      }
    
      input[type="password"] {
        border-top-left-radius: 0;
        border-top-right-radius: 0;
        margin-bottom: 10px;
      }
    }

    .authentication-actions {
      margin: 10px;
  
      a {
        margin: 5px;
        display: block;
      }
    }
  }
}
