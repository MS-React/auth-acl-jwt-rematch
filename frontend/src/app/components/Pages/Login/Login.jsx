import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import LoginForm, { fields } from '../../Common/Form/Templates/Login';
import Actions from '../../../actions';

import './Login.scss';

const Login = ({ login }) => (
  <section className="login-page">
    <div className="login-page--form">
      <div className="login-form">
        <h1 className="h3 mb-3 font-weight-normal">
          Please sign in
        </h1>
        <LoginForm
          fields={fields}
          onSubmit={login}
        />
      </div>
      <div className="authentication-actions">
        <Link to="/signup">Sign Up</Link>
        <Link to="/forgotpassword">Forgot Password</Link>
      </div>
    </div>
  </section>
);

Login.propTypes = {
  login: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  login: formFields => dispatch(Actions.Auth.loginRequest(
    formFields.name.value,
    formFields.password.value,
  )),
});

export default connect(null, mapDispatchToProps)(Login);
