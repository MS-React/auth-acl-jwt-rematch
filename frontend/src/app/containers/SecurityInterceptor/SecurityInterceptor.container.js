import { connect } from 'react-redux';

import Actions from '../../actions';
import SecurityInterceptor from './SecurityInterceptor';

export default connect(
  state => ({
    logged: state.Auth.logged,
    token: localStorage.getItem('jwt-token-id'),
  }), // state
  dispatch => ({
    authenticate: token => dispatch(Actions.Auth.getUserDataByToken(token)),
  }),
)(SecurityInterceptor);
