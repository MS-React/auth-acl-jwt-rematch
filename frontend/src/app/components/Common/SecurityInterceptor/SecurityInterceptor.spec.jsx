import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { Route } from 'react-router';
import Login from '../../Pages/Login';
import SecurityInterceptor from './SecurityInterceptor';

jest.mock('../../Pages/Login');
jest.mock('react-router');

describe('<SecurityInterceptor />', () => {
  let wrapper;
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      children: (<div key="children-key">Children</div>),
      UnauthenticatedComponent: Login,
      authenticate: jest.fn(),
    };
  });

  describe('user not logged', () => {
    beforeEach(() => {
      Login.mockClear();
      wrapper = shallow(<SecurityInterceptor {...defaultProps} />);
    });

    it('should render with the given <UnauthenticatedComponent /> to Login', () => {
      expect(renderer.create(wrapper)
        .toJSON())
        .toMatchSnapshot();
    });
  });

  describe('user is authenticated by Token', () => {
    beforeEach(() => {
      defaultProps.token = 'token-id';
      wrapper = shallow(<SecurityInterceptor {...defaultProps} />);
    });

    it('should call authenticate callback with the token to keep the session', () => {
      expect(defaultProps.authenticate).toHaveBeenCalledWith('token-id');
    });
  });

  describe('user is logged in', () => {
    beforeEach(() => {
      Route.mockClear();
      defaultProps.logged = true;
      defaultProps.path = '/';

      wrapper = shallow(<SecurityInterceptor {...defaultProps} />);
    });

    it('should render with the router given by the path', () => {
      expect(wrapper.find('Route').props()).toEqual(expect.objectContaining({
        path: '/',
      }));
    });
  });
});
