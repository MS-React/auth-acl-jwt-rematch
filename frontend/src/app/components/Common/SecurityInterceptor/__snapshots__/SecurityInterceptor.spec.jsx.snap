// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SecurityInterceptor /> user not logged should render with the given <UnauthenticatedComponent /> to Login 1`] = `null`;
